usr/share/rear/lib/global-functions.sh

```diff
@@ -863,6 +863,62 @@ function umount_mountpoint_lazy() {
+    local what_is_mounted="$2"
+    contains_visible_char "$what_is_mounted" || what_is_mounted="$mountpoint"
+    # First attempt to umount:
+    umount $v "$mountpoint" && return 0
+    # First attempt to umount failed:
+    Log "Failed to umount $what_is_mounted (will retry after one second)"
+    # Normal umounting something directly after some I/O command (like 'cp' above)
+    # may sometimes fail with "target is busy" (cf. 'busy' and 'lazy' in "man umount")
+    # so we retry after one second to increase likelihood that it then succeeds
+    # cf. https://github.com/rear/rear/issues/2908#issuecomment-1382000811 ("sleep 1 works fine")
+    # and https://github.com/rear/rear/issues/3397#issuecomment-2656911018 (sleep also worked here)
+    # because normal umount is preferred over more sophisticated attempts
+    # like lazy umount or enforced umount which raise their own specific troubles
+    # and the -M option for fuser which is used below is not available on older
+    # Linux distributions like RHEL6 and SLES11 so 'sleep 1' and retry is best:
+    sleep 1
+    # Retry the same umount as in the first attempt:
+    umount $v "$mountpoint" && return 0
+    # Retry to umount also failed:
+    Log "Again failed to umount $what_is_mounted"
+    # Show in the log file what still uses the mountpoint:
+    Log "$what_is_mounted is still in use by ('kernel mount' is always there)"
+    # The -M option avoids that fuser may show all processes using the '/' filesystem
+    # e.g. for mountpoint $TMP_DIR/somedir ($TMP_DIR = $BUILD_DIR/tmp = /var/tmp/rear.XXXXXXXXXXXXXXX/tmp/)
+    # when $TMP_DIR/somedir got umounted just before fuser starts, see "man fuser":
+    #   The mount -m option will match any file within the same device as the specified file,
+    #   use the -M option as well if you mean to specify only the mount point.
+    # So when $TMP_DIR/somedir is umounted 'fuser -v -M -m $TMP_DIR/somedir' only shows
+    #   "Specified filename /var/tmp/rear.XXXXXXXXXXXXXXX/tmp/somedir is not a mountpoint"
+    # instead of all processes using '/' (or /var/ or /var/tmp/ if one is a mountpoint)
+    # which would be misleading information that may even look scaring and cause false alarm.
+    # Older systems do not support -M but we must use it to avoid misleading information or false alarm.
+    # Since this code path is exceptional and the output is used only for information and only in the log file
+    # we do not care when fuser fails with "M: unknown signal; fuser -l lists signals":
+    fuser -v -M -m "$mountpoint" 1>&2 || Log "Presumably 'fuser' does not support the -M option"
+    DebugPrint "Trying 'umount --lazy $mountpoint' (normal umount failed)"
+    # Do only plain 'umount --lazy' without additional '--force'
+    # because enforced umount raises its own specific troubles
+    # so we cannot use the umount_mountpoint_lazy() function here:
+    umount $v --lazy "$mountpoint" && return 0
```

Comment:
```
-l, --lazy
           Lazy unmount. Detach the filesystem from the file hierarchy now, and clean up all references to this filesystem as
           soon as it is not busy anymore.

           A system reboot would be expected in near future if you’re going to use this option for network filesystem or
           local filesystem with submounts. The recommended use-case for umount -l is to prevent hangs on shutdown due to an
           unreachable network share where a normal umount will hang due to a downed server or a network partition. Remounts
           of the share will not be possible.
```

which kind of problems to we create for the system? The manpage says that a reboot should follow soonish, but with `rear mkrescue` this is unlikely to happen.

Reply:
I don't know and I don't rememeber all the details of
https://github.com/rear/rear/pull/2909
which is the original implementation
with all the background discussions
wherefrom this one is basically only a copy.